### PR TITLE
Complete the basic mochitest-browser test for the new webconsole-frontend

### LINKED ISSUE #99: Add a waitForMessages function for end-to-end testing
We should see if it's possible to run some of the integration tests by simulating the same `new-messages` event and making changes to head.js without modifying all of the tests.

### PR DESCRIPTION
**Pull request by bgrins**
Fixes #99
## Testing instructions
1. Run `./mach mochitest devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js`

### CODE REVIEW

**Comment by bgrins** [comment]
NOTE: this isn't passing / ready to land right now because of propType errors, so this depends on #150

**Comment by bgrins** [comment]
This will also allow us to do a talos push for #104

**Comment by linclark** [comment]
I had to come back to my computer to say just how excited I am for this!!!

File: devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js [hunk 1]
```diff
@@ -9,15 +9,58 @@ const TEST_URI = "http://example.com/browser/devtools/client/webconsole/new-cons
...
-  // @TODO: fix proptype errors and actually emit 'new-messages' event
-  // let receivedLog = ui.once("new-messages");
-  // yield ContentTask.spawn(gBrowser.selectedBrowser, {}, function() {
-  //   content.wrappedJSObject.doLogs();
-  // });
-  // yield receivedLog;
+  // @TODO: fix proptype errors
+  let receievedMessages = waitForMessages({
+    hud,
+    messages: [{
```
Comment (linclark) [inline devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js:21]: Should each line be an object here?

File: devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js [hunk 2]
```diff
@@ -9,15 +9,58 @@ const TEST_URI = "http://example.com/browser/devtools/client/webconsole/new-cons
...
+    let receivedLog = hud.ui.on("new-messages", function messagesReceieved(e, newMessage) {
+      for (let message of messages) {
+        if (message.matched) {
+          continue;
+        }
+
+        if (newMessage.node.querySelector(".message-body").textContent == message.text) {
+          numMatched++;
+          message.matched = true;
+          info("Matched a message with text: " + message.text + ", still waiting for " + (messages.size - numMatched) + " messages");
```
Comment (linclark) [inline devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js]: I got `Matched a message with text: 2, still waiting for NaN messages`. `messages` is an array and not an immutable here, so should probably be `messages.length`

Comment (bgrins) [inline devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js]: Ah, forgot to fix that up

File: devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js (hunk 1 above)
Comment (bgrins) [inline devtools/client/webconsole/new-console-output/test/mochitest/browser_webconsole_init.js:21]: Yes

**Comment by bgrins** [comment]
Pushed up review fixes

**Comment by linclark** [comment]
I like it 👍

**Pull request merged**